tests/core/test_macros.py

```diff
@@ -27,6 +27,15 @@ def upper_case(evaluator: MacroEvaluator, expression: exp.Condition) -> str:
     def noop(evaluator: MacroEvaluator):
         return None
 
+    @macro("add_ints", coerce_args=True)
+    def add_lits(evaluator: MacroEvaluator, x: int, y: int) -> int:
+        return x + y
```

Comment:
This is a contrived test to show it working, prime use cases are just making the interface more explicit such as in the following test along with bailing on types which are not coercible.